### DISCUSSION TITLE
Scoped CSS in netcoreapp3.1

### DISCUSSION THREAD

**Discussion by stefanloerwald** [dotnet/aspnetcore#27729] [Q&A] [answered]
Is scoped CSS supposed to be active in a project targetting `netcoreapp3.1`? I get build errors, because I have a file `foo.razor.css` without a file `foo.razor`.

To reproduce:

1. dotnet new blazorserver -f netcoreapp3.1
2. echo "" > foo.razor.css
3. dotnet build

> error BLAZOR102: The scoped css file 'foo.razor.css' was defined but no associated razor component was found for it.

There's no build error for a class lib targetting netstandard2.1. Is this on purpose or should it only work net5.0 and following?

**Comment by javiercn** [comment, marked as answer]
Scoped CSS is a 5.0 feature and won't be coming to 3.1

**Reply by stefanloerwald** [reply]
That wasn't exactly what I asked... Maybe more clearly: is this a feature or a bug?

**Reply by javiercn** [reply]
Ah, sorry, I didn't realize that it was a different question (long day).

Scoped CSS is a 5.0 feature and won't be coming to 3.1

**Reply by stefanloerwald** [reply]
Okay, treating this as a bug then. Can discussions be moved to issues?

**Reply by javiercn** [reply]
Unfortunately there is no way to convert discussions into issues, but I'm not sure what the bug here is, can you clarify?

**Reply by stefanloerwald** [reply]
Well given the above instructions, it looks like scoped CSS processing runs when the target framework is set to netcoreapp3.1, even though it should only do so on net5.0+.

**Reply by javiercn** [reply]
Yeah, this is a bug. The issue is here, you can disable manually if you need to `<ScopedCssEnabled>false</ScopedCssEnabled>`
https://github.com/dotnet/aspnetcore/blob/master/src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets#L76

**Reply by stefanloerwald** [reply]
Great that you already found the root cause. Thanks for the hard work!

**Reply by stefanloerwald** [reply]
I created an issue (https://github.com/dotnet/aspnetcore/issues/27768) for this, as it's a more appropriate place to track bugs than here.
